Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -338,41 +354,37 @@ void UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithR
             RELEASE_LOG(UnifiedTextReplacement, "UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange (%s) => exception when cloning contents", uuid.toString().utf8().data());
             return;
         }
+
         m_originalDocumentNodes.set(uuid, contents.returnValue()); // Deep clone.
     }
 
-    auto& replacements = m_replacements.ensure(uuid, [] {
-        return Vector<Replacement> { };
-    }).iterator->value;
-
-    auto [newText, newRange] = [&] {
-        if (replacements.isEmpty())
-            return std::make_tuple(attributedText.string, range);
-
-        auto lastReplacement = replacements.last();
-        auto lastReplacementSize = lastReplacement.attributedText.string.length();
+    RefPtr fragment = WebCore::createFragment(*frame, attributedText.nsAttributedString().get(), WebCore::AddResources::No);
+    if (!fragment) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
 
-        auto newTextResult = attributedText.string.substring(lastReplacementSize);
-        auto newRangeResult = WebCore::CharacterRange { range.location + lastReplacementSize, range.length - lastReplacementSize };
+    replaceContentsInRange(*frame, resolvedRange, *fragment);
 
-        return std::make_tuple(newTextResult, newRangeResult);
-    }();
+    auto selectedTextRange = frame->selection().selection().firstRange();
+    if (!selectedTextRange) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
 
-    auto newResolvedRange = resolveCharacterRange(*sessionRange, newRange);
+    RELEASE_ASSERT(sessionRangeCharacterCount >= range.location + range.length);
```

Comment:
is this release assert ok to leave in? It's needed because otherwise there would be an underflow in the below lines if this was somehow ever false. I could do an early return with a debug assert too, but this should never ever happen in the first place.